### PR TITLE
fix(packagist): Improve fetch concurrency

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

Improve concurrency for fetching multiple auxilary files.

## Context

- Ref: #19495

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
The `AggregateError` described in the #20009 was introduced with this change. However, the error was like this:
- Private package lookup silently failed
- Default `packagist.org` registry was queried, 404 errors were wrapped into `AggregateError` which was reported in the Dependency Dashboard

File: lib/modules/datasource/packagist/index.ts [hunk 1]
```diff
@@ -157,34 +157,35 @@ export class PackagistDatasource extends Datasource {
...
-            includesPackages[key] = dep;
-          }
+        for (const [key, val] of Object.entries(res.packages ?? {})) {
+          includesPackages[key] = PackagistDatasource.extractDepReleases(val);
         }
-      }
+      });
     }
+
+    await p.all(tasks);
```
Comment (rarkins) [inline lib/modules/datasource/packagist/index.ts:187]: No limit?

Comment (zharinov) [reply lib/modules/datasource/packagist/index.ts:187]: No, using default which is `5`

Comment (rarkins) [reply lib/modules/datasource/packagist/index.ts:187]: OK, I wasn't aware it had a default

**Review by rarkins — APPROVED** [review]

File: lib/modules/datasource/packagist/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/datasource/packagist/index.ts:187]: Most frequently used value at the moment factored out p-utils usage into our own util functions.

**Pull request merged**